docs/minhash.rst

```diff
@@ -115,6 +115,41 @@ MurmurHash3: `mmh3 <https://pypi.org/project/mmh3/>`__
 .. figure:: /_static/hashfunc/minhash_benchmark_farmhash.png
    :alt: MinHash Benchmark
 
+GPU usage (experimental)
+------------------------
+
+``MinHash`` can optionally run part of :meth:`MinHash.update_batch` on a CUDA GPU
+via `CuPy <https://cupy.dev/>`_. Hashing and permutation *generation* remain on CPU;
+only the permutation application and min-reduction may use the GPU.
+
+Control behavior with the constructor argument ``gpu_mode``:
+
+- ``'disable'`` (default): always CPU.
+- ``'detect'``: use GPU if available, otherwise fallback to CPU.
+- ``'always'``: require GPU; raises a ``RuntimeError`` if no CUDA device is available.
+
+.. code-block:: python
+
+    from datasketch import MinHash
+
+    m = MinHash(num_perm=256, gpu_mode='detect')
+    m.update_batch([b"token1", b"token2"])
+
+This makes serialized MinHash objects portable: ``'detect'`` will transparently use
+a GPU when present, otherwise CPU. ``'always'`` is useful to enforce GPU execution
+in controlled environments.
+
+GPU Benchmark
+-------------
+Run:
+
+.. code-block:: bash
+
+    python benchmark/sketches/minhash_gpu_benchmark.py \
+      --sizes 1000 10000 50000 --num-perm 128 256 512 --repeats 5 --warmup --plot
+
+This produces CSV to stdout and figures in ``benchmark_out/``.
```

Comment:
Let's skip the benchmark running code block, and include the figures directly. Also include some discussion on the results.

1. runtime comparison for CPU vs GPU over different num_perm, with fixed n
2. runtime comparison for CPU vs GPU over different n, with fixed num_perm.